### PR TITLE
feat(core): added ignore click on selection directive

### LINKED ISSUE #7910: Dynamic Page issue with copy clickable header area
#### Is this a bug, enhancement, or feature request?
bug

#### Briefly describe your proposal.
Allow the select text from the clickable header area in Dynamic Page

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
latest

#### If this is a bug, please provide steps for reproducing it.
1. go to https://fundamental-ngx.netlify.app/#/core/dynamic-page#tabbed
2. click on the button for `Default tabs example`
3. try to select text from the header - you will see that instead of having the text selected the subheader collapses/expands

dxp/core/issues/483

### PR DESCRIPTION
**Pull request by g-cheishvili**
## Related Issue(s)

closes #7910 

## Description

Added directive, which will listen on element click and if document has active selection from that element it will not propagate click events. This way we do not have to cancel every event, but only the ones which result selection

### CODE REVIEW

**Review by droshev — APPROVED** [review]

File: libs/core/src/lib/utils/directives/ignore-click-on-selection/ignore-click-on-selection.directive.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            provide: IgnoreClickOnSelectionDirectiveToken,
+            useExisting: IgnoreClickOnSelectionDirective
+        }
+    ]
+})
+export class IgnoreClickOnSelectionDirective {
+    constructor(@Inject(DOCUMENT) private document: Document, private _elementRef: ElementRef<HTMLElement>) {}
+
+    @HostListener('click', ['$event'])
```
Comment (InnaAtanasova) [inline libs/core/src/lib/utils/directives/ignore-click-on-selection/ignore-click-on-selection.directive.ts:17]: missing typedocs

File: libs/core/src/lib/utils/directives/ignore-click-on-selection/ignore-click-on-selection.directive.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                this._elementRef.nativeElement.isSameNode(selection.anchorNode) ||
+                this._elementRef.nativeElement.isSameNode(selection.anchorNode.parentElement)
+            ) {
+                $event.stopPropagation();
+                $event.preventDefault();
+            }
+        }
+    }
+
+    nativeElement(): Element {
```
Comment (InnaAtanasova) [inline libs/core/src/lib/utils/directives/ignore-click-on-selection/ignore-click-on-selection.directive.ts:31]: missing typedocs

File: libs/core/src/lib/utils/directives/ignore-click-on-selection/ignore-click-on-selection.directive.ts (hunk 1 above)
Comment (g-cheishvili) [reply libs/core/src/lib/utils/directives/ignore-click-on-selection/ignore-click-on-selection.directive.ts:17]: dammit

**Comment by InnaAtanasova** [comment]
Please maintain the right-hand section of the PR - labels, project, milestone, assignee...

**Pull request merged**